libraries/AP_Avoidance/AP_Avoidance.cpp

```diff
@@ -593,14 +593,17 @@ void AP_Avoidance::handle_msg(const mavlink_message_t &msg)
     // inform AP_Avoidance we have a new player
     mavlink_global_position_int_t packet;
     mavlink_msg_global_position_int_decode(&msg, &packet);
-    Location loc;
-    loc.lat = packet.lat;
-    loc.lng = packet.lon;
-    loc.alt = packet.alt / 10; // mm -> cm
-    loc.relative_alt = false;
-    Vector3f vel = Vector3f(packet.vx/100.0f, // cm to m
-                            packet.vy/100.0f,
-                            packet.vz/100.0f);
+    const Location loc {
+        packet.lat,
+        packet.lon,
+        int32_t(packet.alt * 0.01),  // mm -> cm
+        Location::AltFrame::ABSOLUTE
+    };
+    const Vector3f vel {
+        packet.vx * 0.01f, // cm to m
+        packet.vy * 0.01f,
+        packet.vz * 0.01f
```

Comment:
```suggestion
    const Vector3f vel {
        packet.vx * 0.01, // cm to m
        packet.vy * 0.01,
        packet.vz * 0.01
```

Reply:
I've removed these - but I think that's going to cause a bunch of narrowing warnings - so I'll probably re-add them in that case.

Reply:
Re-added them.